README.md

```diff
@@ -41,6 +41,12 @@ This include all tests under test/
 
 The client should be as simple as is practical to implement the desired user experience.
 
+* Pages should only include the JavaScript needed to implement their own behavior.
+* Client-side JavaScript should have no need to manage browser history or derive state from the URL.
+* Client-side JavaScript should avoid API-driven interactions, and instead access JSON embedded in the initial page load and submit data using HTML forms.
+
+For example, enable/disable logic in forms can be specified server-side with HTML [data attributes](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) such as `data-disabled-if-parent="parent-element-name" data-disabled-if-parent-less-than="10"` and then implemented with generic client-side JS that responds to DOM events relevant to the attribute-specified elements.
```

Comment:
I think I get what you're getting at with the server-side enable/disable logic, but it's not 100% clear to me just now. Anyway, I assume it'll be come clearer once we start implementing this logic.